### PR TITLE
Use mmap option to load_state_dict

### PR DESCRIPTION
**Pull request by weimingzha0**
# Use torch.load(mmap=True) to accelerate checkpoint loading

https://github.com/huggingface/transformers/issues/28332

cc @SunMarc @sgugger

### CODE REVIEW

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -515,8 +516,15 @@ def load_state_dict(checkpoint_file: Union[str, os.PathLike]):
             map_location = "meta"
         else:
             map_location = "cpu"
-
-        return torch.load(checkpoint_file, map_location=map_location, weights_only=True)
+        extra_args = {}
+        # mmap can only be used with files serialized with zipfile-based format.
```
Comment (ArthurZucker) [inline src/transformers/modeling_utils.py:520]: how often does this come up? (zipfile-based format)

**Review by ArthurZucker — COMMENTED** [review]
torch bin files are now deprecated in favor of safetensors but no harm in improving this! 
Coul you add a test as well in test_modelling_common? 🤗

**Comment by weimingzha0** [comment]
> torch bin files are now deprecated in favor of safetensors but no harm in improving this! Coul you add a test as well in test_modelling_common? 🤗

Sure. Please let me know if my test case is correct.

**Review by ArthurZucker — APPROVED** [review]
Thanks! My last query would be to make sure this works for the deepspeed / sdpa case (device_map = "meta")!

**Comment by weimingzha0** [comment]
> Thanks! My last query would be to make sure this works for the deepspeed / sdpa case (device_map = "meta")!

Do you mean add a test case for device_map == "meta' ?

**Comment by ArthurZucker** [comment]
That would be a good way of making sure this will be fine with DeepSpeedZero (the if branch) 🤗

**Comment by weimingzha0** [comment]
> That would be a good way of making sure this will be fine with DeepSpeedZero (the if branch) 🤗

Currently none of existing tests covers the branch: 
I even added an assert in the "if branch" and all tests still passed except for one irrelevant error (see https://github.com/huggingface/transformers/pull/28401)

Anyway, I added a guard for "meta" device.

**Review by ArthurZucker — APPROVED** [review]
Good, thanks for iterating

**Pull request merged**